src/fastgltf.cpp

```diff
@@ -5711,13 +6258,368 @@ void fg::Exporter::writeTextures(const Asset& asset, std::string& json) {
+				json += R"(")";
+				json += *systemIt;
+				json += R"(")";
+
+				if (uabs(std::distance(filter.collideWithSystems.begin(), systemIt)) + 1 < filter.collideWithSystems.size()) {
+					json += ',';
+				}
+			}
+			json += "]";
+		}
+		if (json.back() == ']') {
+			json += ',';
+		}
+
+		if (!filter.notCollideWithSystems.empty()) {
+			json += R"("notCollideWithSystems":[)";
+			for (auto systemIt = filter.notCollideWithSystems.begin(); systemIt != filter.notCollideWithSystems.end(); ++systemIt) {
+				json += R"(")";
+				json += *systemIt;
+				json += R"(")";
+
+				if (uabs(std::distance(filter.notCollideWithSystems.begin(), systemIt)) + 1 < filter.notCollideWithSystems.size()) {
+					json += ',';
+				}
+			}
+			json += "]";
+		}
+
+		if (extrasWriteCallback != nullptr) {
+			auto extras = extrasWriteCallback(uabs(std::distance(asset.collisionFilters.begin(), it)), fastgltf::Category::Shapes, userPointer);
+			if (extras.has_value()) {
+				if (json.back() != '{') {
+					json += ',';
+				}
+				json += std::string("\"extras\":") + *extras;
+			}
+		}
+
+		json += "}";
+
+		if (uabs(std::distance(asset.collisionFilters.begin(), it)) + 1 < asset.collisionFilters.size()) {
+			json += ',';
+		}
+	}
+
+	json += "]";
+}
+
+void fg::Exporter::writePhysicsJoints(const Asset& asset, std::string& json) {
+	if (asset.physicsJoints.empty()) {
+		return;
+	}
+	if (json.back() == ']' || json.back() == '}') {
+		json += ',';
+	}
+
+	json += R"("physicsJoints":[)";
+
+	for (auto it = asset.physicsJoints.begin(); it != asset.physicsJoints.end(); ++it) {
+		const auto& joint = *it;
+
+		json += "{";
+
+		if (!joint.limits.empty()) {
+			json += R"("limits":[)";
+			for (auto limitIt = joint.limits.begin(); limitIt != joint.limits.end(); ++limitIt) {
+				const auto& limit = *limitIt;
+
+				json += "{";
+				if (!limit.linearAxes.empty()) {
+					json += R"("linearAxes":[)";
+					for (auto axisIt = limit.linearAxes.begin(); axisIt != limit.linearAxes.end(); ++axisIt) {
+						json += std::to_string(*axisIt);
+
+						if (uabs(std::distance(limit.linearAxes.begin(), axisIt)) + 1 < limit.linearAxes.size()) {
+							json += ',';
+						}
+					}
+					json += "],";
+				}
+				if (!limit.angularAxes.empty()) {
+					json += R"("angularAxes":[)";
+					for (auto axisIt = limit.angularAxes.begin(); axisIt != limit.angularAxes.end(); ++axisIt) {
+						json += std::to_string(*axisIt);
+
+						if (uabs(std::distance(limit.angularAxes.begin(), axisIt)) + 1 < limit.angularAxes.size()) {
+							json += ',';
+						}
+					}
+					json += "],";
+				}
+				if (limit.min) {
+					json += R"("min":)" + std::to_string(*limit.min) + ",";
+				}
+				if (limit.max) {
+					json += R"("max":)" + std::to_string(*limit.max) + ",";
+				}
+				if (limit.stiffness) {
+					json += R"("stiffness":)" + std::to_string(*limit.stiffness) + ",";
+				}
+			    json += R"("damping":)" + std::to_string(limit.damping);
```

Comment:
These should all be `to_string_fp` since `std::to_string(float)` doesn't have round-trip guarantees.